src/test/regress/expected/ensure_no_intermediate_data_leak.out

```diff
@@ -0,0 +1,17 @@
+------
+-- THIS TEST SHOULD BE EXECUTED AT THE END OF THE REGRESSION TEST SUITE
```

Comment:
I'll update this comment. 

@marcocitus do you have any other comments before merging this PR?

I'll open an issue describing the cases where Citus fails to clear all intermediate results:
- Failures in task-tracker
- Windows (not sure about the source)